float8_experimental/float8_dynamic_linear.py

```diff
@@ -54,33 +75,30 @@ def __init__(self, **super_kwargs):
         super().__init__(**super_kwargs)
 
     def forward(self, x):
-        # cast x to float8_e4m3fn if not using activation hooks
         x_fp8 = self.cast_to_float8_e4m3fn(x)
-
-        # cast w to float8_e4m3fn
-        w_fp8 = self.cast_to_float8_e4m3fn(self.weight)
-
+        w_fp8 = (
```

Comment:
optional nit: could just use plain if/else for readability? It will be 4 LOC, same as this syntax, but less mental load to understand